src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,44 +40,97 @@ define([
     };
 
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
-        this._widget.setExecutionDisplayed = this.setExecutionDisplayed.bind(this);
+        this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
+    };
+
+    ExecutionIndexControl.prototype.setDisplayedExecutions = function (displayedIds) {
+        this.displayedExecutions = displayedIds;
+        this._updateGraphWidget();
     };
 
-    ExecutionIndexControl.prototype.setExecutionDisplayed = function (id, bool) {
-        var lines = this._linesForExecution[id] || [],
-            otherLines,
-            wasMultiLine = this.displayedExecCount() > 1,
-            isMultiLine;
-
-        this._logger.info(`setting execution ${id} to ${bool ? 'displayed' : 'hidden'}`);
-        this.displayedExecutions[id] = bool;
-
-        // If we just crossed the multi line threshold, then update all the lines
-        isMultiLine = this.displayedExecCount() > 1;
-        if (isMultiLine !== wasMultiLine) {
-            // Refresh the other lines visible
-            otherLines = Object.keys(this.displayedExecutions)
-                .filter(eId => this.displayedExecutions[eId] && (eId !== id))
-                .map(id => this._linesForExecution[id] || [])
-                .reduce((l1, l2) => l1.concat(l2), []);
-
-            this._updateLines(otherLines, false);
-            this._updateLines(otherLines, true);
+    ExecutionIndexControl.prototype._updateGraphWidget = function () {
+        const action = this.displayedExecCount() === 0 ? 'addNode' : 'updateNode';
```

Comment:
This is no longer checking what it was prior to the last refactor. Previously, this check served to identify when the graph changed from no nodes to having nodes. However, now the check is happening after the execution count is updated. For example, suppose we are changing from showing 0 executions to 1. In this case, `updateNode` will be used rather than `addNode` since the execution count has already been updated (line 47). 

Do we even need to differentiate between `addNode` and `updateNode` or can we just make a generic `update` method for the widget?

Reply:
Actually its not necessary to differenciate as it relates to the widget, the logic in the widget differenciates between the `plotly.newPlot` and `plotly.react`. This can be removed and the widget can be called based on a generic `updateMethod`.

Reply:
Now the calls are only to `updateNode` which will take care of creating and updating the plotly plot.